feature.go

```diff
@@ -58,3 +60,65 @@ func (f *Feature) Properties() map[string]interface{} {
 
 	return props
 }
+
+func (f *Feature) Unmarshal(v interface{}) error {
+	rv := reflect.ValueOf(v)
+
+	if rv.Kind() != reflect.Ptr || rv.IsNil() || rv.Elem().Kind() != reflect.Struct {
+		return fmt.Errorf("cannot unmarshal")
+	}
+
+	// TODO: handle error
+	geometry, _ := f.Geometry()
+	props := f.Properties()
+
+	rt := rv.Elem().Type()
+	gt := reflect.TypeOf(geometry)
+	for i := 0; i < rt.NumField(); i++ {
+		field := rt.Field(i)
+		fieldType := field.Type
+		fieldKind := fieldType.Kind()
+
+		target := field.Name
+		tag := field.Tag.Get("fgb")
+		if tag != "" {
+			target = tag
+		}
+
+		target = strings.ToLower(target)
+
+		if target == "geom" {
+			var geomInterface geom.T
+
+			if fieldKind != reflect.Ptr && fieldType != reflect.TypeOf(&geomInterface).Elem() {
```

Comment:
You can remove the variable and rewrite the condition as `reflect.TypeOf((*geom.T)(nil))`